docs/user-guide/lns-discovery.md

```diff
@@ -0,0 +1,65 @@
+# LNS discovery
+
+The LNS protocol specifies that the first request an LBS makes is to the `/router-info` discovery endpoint. The LNS from the starter kit implements the `/router-info` endpoint and redirects any requests to that endpoint forward to the `/router-data` endpoint on the same LNS.
+
+In addition to the discovery endpoint on the LNS itself, we also support a standalone discovery service, which you can use to distribute LBS connections to different LNS. The discovery service is a .NET application, and its implementation is part of the `LoRaWan.NetworkServerDiscovery` project.
+
+The discovery service relies on configuration from IoT Hub module twins to associate each LBS with a set of LNS to which it may connect to. For this the configuration needs to take into account possible network boundaries - by defining these networks you can control which LBS should connect to which LNS. For this you need to configure your LBS twin in IoT Hub as follows:
+
+```json
+{
+  ...,
+  "tags": {
+    "network": "<network-id>"
+  },
+  ...
+}
+```
+
+The network ID may only consist of alphanumeric characters. After you configured the LBS, you need to also configure your LNS with the same network. The LNS twin furthermore needs to contain the IP/DNS of the host, hence the twin of the LNS *module* should look like:
+
+```json
+{
+  ...,
+  "tags": {
+    "network": "<network-id>"
+  },
+  ...,
+  "properties": {
+    "desired": {
+      "hostAddress": "wss://mylns:5001",
+      ...
+    }
+  }
+}
+```
+
+Keep in mind that the `hostAddress` value must include the scheme and the host name. In case you omit the port, the defaults of the protocol will be used.
+
+!!! note
+    The discovery endpoint distributes LBS to LNS using a round-robin distribution mechanism. It will always try the same LNS per LBS first. Hence, it can be used for an active/passive scenario, but not to distribute load more evenly across different LNS.
+
+!!! note
+    The configuration values of the network name of a station, respectively of the set of LNS in a given network, are cached by the discovery endpoint for six hours.
+    If you updated the configuration, make sure to restart the discovery service to ensure that the cache is refreshed.
+
+You can choose to deploy the discovery service either on-prem or in the cloud. For both deployment strategies, you can configure the following behavior:
+
+- The log level can be configured by setting the `Logging__LogLevel__Default` environment variable. For more fine-grained configuration of the console log level, refer to [Logging in .NET Core and ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-6.0#set-log-level-by-command-line-environment-variables-and-other-configuration).
```

Comment:
it would be great to have a table with configuration settings that are supported, a description, supported values and the default we apply, if not specified.

Reply:
Good idea, done in 3debf30343125d264b71101fe0e66cb71967a5fe